content/part2.md

```diff
@@ -0,0 +1,186 @@
+
+With a traditional web app, abandoned cart notifications are tricky.
+You need to use a job queue like [Celery](https://en.wikipedia.org/wiki/Celery_(software\)) in Python or [Machinery](https://github.com/RichardKnop/machinery) in Golang.
+You need to schedule a job that checks if the cart is abandoned, and reschedule that job every time the cart is updated.
+
+With Temporal, you don't need a separate job queue. Instead, you define a _Selector_ with two event handlers: one that responds to a Workflow signal, and one that responds to a timer.
+By creating a new Selector on each iteration of the `for` loop, you're telling Temporal to handle the next update cart signal it receives, or send an abandoned cart email if it doesn't receive a signal for `abandonedCartTimeout`.
+Calling `Select()` on a selector blocks the Workflow until there's either a signal or `abandonedCartTimeout` elapses.
+
+```golang
+func CartWorkflow(ctx workflow.Context, state CartState) error {
+	logger := workflow.GetLogger(ctx)
+
+	err := workflow.SetQueryHandler(ctx, "getCart", func(input []byte) (CartState, error) {
+		return state, nil
+	})
+	if err != nil {
+		logger.Info("SetQueryHandler failed.", "Error", err)
+		return err
+	}
+
+	channel := workflow.GetSignalChannel(ctx, "cartMessages"
+	sentAbandonedCartEmail := false
+
+	for {
+    // Create a new selector on each iteration of the loop means Temporal will pick the first
+    // event that occurs each time: either receiving a signal, or responding to the timer.
+		selector := workflow.NewSelector(ctx)
+		selector.AddReceive(channel, func(c workflow.ReceiveChannel, _ bool) {
+			var signal interface{}
+			c.Receive(ctx, &signal)
+
+			// Handle signals for updating the cart
+		})
+
+    // If the user doesn't update the cart for `abandonedCartTimeout`, send an email
+    // reminding them about their cart. Only send the email once.
+		if !sentAbandonedCartEmail && len(state.Items) > 0 {
+			selector.AddFuture(workflow.NewTimer(ctx, abandonedCartTimeout), func(f workflow.Future) {
+				sentAbandonedCartEmail = true
+				ao := workflow.ActivityOptions{
+					ScheduleToStartTimeout: time.Minute,
+					StartToCloseTimeout:    time.Minute,
+				}
+
+				ctx = workflow.WithActivityOptions(ctx, ao)
+
+        // More on SendAbandonedCartEmail in the next section
+				err := workflow.ExecuteActivity(ctx, SendAbandonedCartEmail, state.Email).Get(ctx, nil)
+				if err != nil {
+					logger.Error("Error sending email %v", err)
+					return
+				}
+			})
+		}
+
+		selector.Select(ctx)
+	}
+
+	return nil
+}
+```
+
+Temporal Selectors make implementing an abandoned cart email trivial.
+No need to implement a job queue, write a separate worker, or handle rescheduling jobs.
+Just create a new Selector after every signal and use `AddFuture()` to handle the case where the user abandons their cart.
+Temporal does the hard work of persisting and distributing the state of your Workflow for you.
```

Comment:
making small clarification that selectors are a Go SDK concept, node a Temporal-wide concept

```suggestion
Temporal's [GO SDK Selectors](https://docs.temporal.io/docs/go/selectors/) make implementing an abandoned cart email trivial.
No need to implement a job queue, write a separate worker, or handle rescheduling jobs.
Just create a new Selector after every signal and use `AddFuture()` to handle the case where the user abandons their cart.
Temporal does the hard work of persisting and distributing the state of your Workflow for you.
```